config/allowlist.go

```diff
@@ -135,3 +230,46 @@ func (a *Allowlist) ContainsStopWord(s string) (bool, string) {
 	}
 	return false, ""
 }
+
+// ExpressionAllowed returns the result of the predicate expression.
+func (a *Allowlist) ExpressionAllowed(
+	// Miserable workaround to import-cycles.
```

Comment:
hmmm maybe it's time `finding` got its own package. Outside of the scope for this PR though